examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb

```diff
@@ -0,0 +1,2163 @@
+        "                      autoscaling_algorithm='NONE',\n",
+        "                      num_workers=10))\n",
+        "\n",
+        "# Create a deferred Beam DataFrame with the contents of our csv file.\n",
+        "beam_df = p | beam.dataframe.io.read_csv(source_csv_file, splittable=True)\n",
+        "\n",
+        "# Drop irrelavant columns/columns with missing values\n",
+        "beam_df = beam_df.drop(['spk_id', 'full_name','diameter', 'albedo', 'diameter_sigma'], axis='columns', inplace=False)\n",
+        "\n",
+        "# Get numerical columns/columns with categorical variables\n",
+        "numerical_cols = beam_df.select_dtypes(include=np.number).columns.tolist()\n",
+        "categorical_cols = list(set(beam_df.columns) - set(numerical_cols))\n",
+        "\n",
+        "# Normalize the numerical variables \n",
+        "beam_df_numerical = beam_df.filter(items=numerical_cols)\n",
+        "beam_df_numerical = (beam_df_numerical - beam_df_numerical.mean())/beam_df_numerical.std()\n",
+        "\n",
+        "# One-hot encode the categorical variables \n",
+        "beam_df_categorical = beam_df.filter(items=categorical_cols)\n",
+        "# Todo: one hot-encoding step\n",
+        "\n",
+        "# Merge the normalized variables with the one-hot encoded variables  (Optional)\n",
+        "preprocessed_dataset = beam_df_categorical.merge(beam_numerical_cols, left_index = True, right_index = True)\n",
+        "\n",
+        "# Write the pre-processed dataset to csv\n",
+        "preprocessed_dataset.to_csv(os.path.join(OUTPUT_DIR, \"preprocessed_data.csv\"))"
+      ],
+      "metadata": {
+        "id": "1XovR0gKbMlK"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Let's now submit and execute our pipeline."
+      ],
+      "metadata": {
+        "id": "a789u4Yecs_g"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "p.run().wait_until_finish()"
+      ],
+      "metadata": {
+        "id": "pbUlC102bPaZ"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "The execution of the pipeline job will take some time until it finishes."
+      ],
+      "metadata": {
+        "id": "dzdqmzKzTOng"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# What's next \n",
+        "\n",
+        "Now that we've seen how we can analyze and preprocess a large-scale dataset with the Beam DataFrames API, we can now train a model on a classification task on our preprocessed dataset.  \n",
+        "\n",
+        "To learn more on how to get started with classifying structured data, refer to:\n",
+        "\n",
+        "*   [Classify structured data with feature columns](https://www.tensorflow.org/tutorials/structured_data/feature_columns)\n",
```

Comment:
Feature columns are no longer recommended. They were used for tf.Estimators, but now that TF2 has moved to Keras they now recommend the preprocessing layers.

The preprocessing layers already take care of normalizing with the z-score and one hot encodings, but it's still nice seeing how to do that with Beam.

Reply:
good to know, i'll remove the resource

Reply:
Good to know. I'll remove the reference